cg/models/demultiplex/flow_cell.py

```diff
@@ -126,6 +126,28 @@ def sequencer_type(
         """Return the sequencer type."""
         return sequencer_types[self.machine_name]
 
+    def set_bcl_converter(self, bcl_converter: str) -> str:
+        """
+        Return the bcl converter to use.
+        Tries to fetch the bcl converter from the sequencer type if not provided.
+        """
+        if bcl_converter:
+            return bcl_converter
+        return self.get_bcl_converter_by_sequencer()
+
+    def get_bcl_converter_by_sequencer(
+        self,
+    ) -> str:
+        """Return the bcl converter based on sequencer."""
+        try:
+            if self.sequencer_type in [Sequencers.HISEQGA, Sequencers.HISEQX]:
+                return BclConverter.BCL2FASTQ
+        except KeyError:
+            raise FlowCellError(
+                f"Could not determine bcl converter from sequencer type for machine name {self.machine_name}"
+            )
+        return BclConverter.DRAGEN
```

Comment:
Wouldn't it be better if you check first if it is hiseq, then check if is novaseq and if not, raise an error? I think this error-catching may not be necessary.

Reply:
Yes, looking at it know the raised error is a bit pointless.